src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -75,9 +75,17 @@ private unsafe static int CompareOrdinalIgnoreCaseHelper(String strA, String str
 
         private unsafe static bool EqualsHelper(String strA, String strB)
         {
-            Contract.Requires(strA != null);
-            Contract.Requires(strB != null);
-            Contract.Requires(strA.Length == strB.Length);
+            Debug.Assert((strA == null || strB == null) || strA.Length == strB.Length);
+
+            if ((object)strA == strB)
```

Comment:
Why do you think moving it here is a good idea?

IIRC, it was moved up as the first check years ago because of somebody collected data that it is the right tradeoff.